File: DynamicKey/AgoraDynamicKey/java/src/main/java/io/agora/sample/RtcTokenBuilder2Sample.java
package io.agora.sample;

import io.agora.media.RtcTokenBuilder2;
import io.agora.media.RtcTokenBuilder2.Role;

public class RtcTokenBuilder2Sample {
    // Need to set environment variable AGORA_APP_ID
    static String appId = System.getenv("AGORA_APP_ID");
    // Need to set environment variable AGORA_APP_CERTIFICATE
    static String appCertificate = System.getenv("AGORA_APP_CERTIFICATE");

    static String channelName = "7d72365eb983485397e3e3f9d460bdda";
    static String account = "2082341273";
    static int uid = 2082341273;
    static int tokenExpirationInSeconds = 3600;
    static int privilegeExpirationInSeconds = 3600;
    static int joinChannelPrivilegeExpireInSeconds = 3600;
    static int pubAudioPrivilegeExpireInSeconds = 3600;
    static int pubVideoPrivilegeExpireInSeconds = 3600;
    static int pubDataStreamPrivilegeExpireInSeconds = 3600;

    public static void main(String[] args) {
        System.out.printf("App Id: %s\n", appId);
        System.out.printf("App Certificate: %s\n", appCertificate);
        if (appId == null || appId.isEmpty() || appCertificate == null || appCertificate.isEmpty()) {
            System.out.printf("Need to set environment variable AGORA_APP_ID and AGORA_APP_CERTIFICATE\n");
            return;
        }

        RtcTokenBuilder2 token = new RtcTokenBuilder2();
        String result =
                token.buildTokenWithUid(appId, appCertificate, channelName, uid, Role.ROLE_PUBLISHER, tokenExpirationInSeconds, privilegeExpirationInSeconds);
        System.out.printf("Token with uid: %s\n", result);

        result = token.buildTokenWithUserAccount(appId, appCertificate, channelName, account, Role.ROLE_PUBLISHER, tokenExpirationInSeconds,
                privilegeExpirationInSeconds);
        System.out.printf("Token with account: %s\n", result);

        result = token.buildTokenWithUid(appId, appCertificate, channelName, uid, tokenExpirationInSeconds, joinChannelPrivilegeExpireInSeconds,
                pubAudioPrivilegeExpireInSeconds, pubVideoPrivilegeExpireInSeconds, pubDataStreamPrivilegeExpireInSeconds);
        System.out.printf("Token with uid and privilege: %s\n", result);

        result = token.buildTokenWithUserAccount(appId, appCertificate, channelName, account, tokenExpirationInSeconds, joinChannelPrivilegeExpireInSeconds,
                pubAudioPrivilegeExpireInSeconds, pubVideoPrivilegeExpireInSeconds, pubDataStreamPrivilegeExpireInSeconds);
        System.out.printf("Token with account and privilege: %s\n", result);

        result = token.buildTokenWithRtm(appId, appCertificate, channelName, account, Role.ROLE_PUBLISHER, tokenExpirationInSeconds,
                privilegeExpirationInSeconds);
        System.out.printf("Token with RTM: %s\n", result);

    }
}

File: DynamicKey/AgoraDynamicKey/kotlin/src/sample/kotlin/io/agora/sample/RtcTokenBuilder2Sample.kt
package io.agora.sample

import io.agora.media.RtcTokenBuilder2

/** Demonstrates RTC Token007 generation with UID and user account identities. */
object RtcTokenBuilder2Sample {
    // Need to set environment variable AGORA_APP_ID
    private val appId = System.getenv("AGORA_APP_ID")

    // Need to set environment variable AGORA_APP_CERTIFICATE
    private val appCertificate = System.getenv("AGORA_APP_CERTIFICATE")

    private const val channelName = "7d72365eb983485397e3e3f9d460bdda"
    private const val account = "2082341273"
    private const val uid = 2082341273
    private const val tokenExpirationInSeconds = 3600
    private const val privilegeExpirationInSeconds = 3600
    private const val joinChannelPrivilegeExpireInSeconds = 3600
    private const val pubAudioPrivilegeExpireInSeconds = 3600
    private const val pubVideoPrivilegeExpireInSeconds = 3600
    private const val pubDataStreamPrivilegeExpireInSeconds = 3600

    /** Generates and prints RTC sample tokens using environment-provided credentials. */
    @JvmStatic
    fun main(args: Array<String>) {
        println("App Id: $appId")
        println("App Certificate: $appCertificate")
        if (appId.isNullOrEmpty() || appCertificate.isNullOrEmpty()) {
            println("Need to set environment variable AGORA_APP_ID and AGORA_APP_CERTIFICATE")
            return
        }

        val tokenBuilder = RtcTokenBuilder2()
        var result = tokenBuilder.buildTokenWithUid(
            appId, appCertificate, channelName, uid, RtcTokenBuilder2.Role.ROLE_PUBLISHER,
            tokenExpirationInSeconds, privilegeExpirationInSeconds
        )
        println("Token with uid: $result")

        result = tokenBuilder.buildTokenWithUserAccount(
            appId, appCertificate, channelName, account,
            RtcTokenBuilder2.Role.ROLE_PUBLISHER,
            tokenExpirationInSeconds, privilegeExpirationInSeconds
        )
        println("Token with account: $result")

        result = tokenBuilder.buildTokenWithUid(
            appId, appCertificate, channelName, uid, tokenExpirationInSeconds,
            joinChannelPrivilegeExpireInSeconds, pubAudioPrivilegeExpireInSeconds,
            pubVideoPrivilegeExpireInSeconds,
            pubDataStreamPrivilegeExpireInSeconds
        )
        println("Token with uid and privilege: $result")

        result = tokenBuilder.buildTokenWithUserAccount(
            appId, appCertificate, channelName, account,
            tokenExpirationInSeconds,
            joinChannelPrivilegeExpireInSeconds, pubAudioPrivilegeExpireInSeconds,
            pubVideoPrivilegeExpireInSeconds, pubDataStreamPrivilegeExpireInSeconds
        )
        println("Token with account and privilege: $result")
    }
}
